### DISCUSSION TITLE
一直很好奇，是否有dns缓存？

### DISCUSSION THREAD

**Discussion by Travel2Here** [Openwrt-Passwall/openwrt-passwall2#303] [Q&A]
Tue Jul 25 09:47:00 2023 daemon.info dnsmasq[18502]: started, version 2.88 cache disabled
Tue Jul 25 09:47:00 2023 daemon.info dnsmasq[18502]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset no-nftset no-auth no-cryptohash no-DNSSEC loop-detect inotify dumpfile
Tue Jul 25 09:47:00 2023 daemon.info dnsmasq[18502]: using nameserver 127.0.0.1#11302
Tue Jul 25 09:47:00 2023 daemon.info dnsmasq[18502]: read /etc/hosts - 6 names

有人说xray有缓存机制，可是在哪里体现？我没找到。如这个日志显示，关闭了dnsmasq的缓存，是否表示每次请求是完全无缓存？
我有试过将app.sh文件里的缓存参数0改为10000，日常使用没有访问加载问题。

另外就是fakedns，其实看介绍逻辑还是挺有用，但在某些app使用中表现为过一段时间就不能正常访问了。比如instagram。

以上。求解。

**Comment by hululu1068** [comment]
有缓存，且统一TTL为600ms。
尝试过关掉缓存，效果没有开启好。
